code/core/src/manager/container/Menu.tsx

```diff
@@ -49,16 +57,21 @@ export const Shortcut: FC<{ keys: string[] }> = ({ keys }) => (
   </Key>
 );
 
-export const useMenu = (
-  state: State,
-  api: API,
-  showToolbar: boolean,
-  isFullscreen: boolean,
-  isPanelShown: boolean,
-  isNavShown: boolean,
-  enableShortcuts: boolean
-): Link[][] => {
+export const useMenu = ({
+  api,
+  showToolbar,
+  isPanelShown,
+  isNavShown,
+  enableShortcuts,
+}: {
+  api: API;
+  showToolbar: boolean;
+  isPanelShown: boolean;
+  isNavShown: boolean;
+  enableShortcuts: boolean;
+}): NormalLink[][] => {
   const shortcutKeys = api.getShortcutKeys();
+  const { progress } = useChecklist();
```

Comment:
This should likely be passed in as a `isGuideProgress` property on the `useMenu` hooks' options.

That we it's not invalidated for every single state change ever in the checkList.